ports/raspberrypi/common-hal/pulseio/PulseIn.c

```diff
@@ -140,14 +140,19 @@ void common_hal_pulseio_pulsein_interrupt() {
             result = level_count;
             last_level = level;
             level_count = 1;
-            // ignore pulses that are too long and too short
-            if (result < MAX_PULSE && result > MIN_PULSE) {
-                self->buffer[buf_index] = result;
+	    // Pulses that are londger than MAX_PULSE will return MAX_PULSE
+	    if (result > MAX_PULSE ) {
+	        result = MAX_PULSE;
+	    }
+            // ignore pulses that are too short
+            if (result <= MAX_PULSE && result > MIN_PULSE) {
```

Comment:
I wonder, why there're hard-coded boundaries in the first place? As for the `MAX_PULSE`, my wild guess is that there're some dependencies on the buffer type which is `uint16_t[]`, is that correct? But for the `MIN_PULSE`, why would that be ever needed?

And I think, this should be stated in the API docs, may save a lot of time for someone.